cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2717,19 +2694,26 @@ impl MachInst for Inst {
                         || ty == types::R64
                 );
                 if value == 0 {
```

Comment:
This branch kind of jumped out at me. Both sides do similar things with OperandSize, but the if 0 case only widens I64 types (not B64 or R64).

I left it as is to keep this a pure refactor, but it struck me as odd.

Reply:
Hmm, yes, good catch -- I think that for consistency this should match the widening logic in the `value != 0` case below. (It's harmless in practice as the 32-bit `xor` will zero the top 32 bits anyway, but we might as well be consistent.)

Reply:
Updated.